src/profile-logic/import/instruments/BinaryPlistParser.js

```diff
@@ -0,0 +1,221 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+export class UID {
+  index: number;
+
+  constructor(index: number) {
+    this.index = index;
+  }
+}
+
+class BinaryPlistParser {
+  view: DataView;
+  referenceSize: number;
+  offsetTable: Array<number>;
+
+  constructor(view: DataView) {
+    this.view = view;
+    this.referenceSize = 0;
+    this.offsetTable = [];
+  }
+
+  parseRoot(): any {
+    const trailer = this.view.byteLength - 32;
+    const offsetSize = this.view.getUint8(trailer + 6);
+    this.referenceSize = this.view.getUint8(trailer + 7);
+
+    // Just use the last 32-bits of these 64-bit big-endian values
+    const objectCount = this.view.getUint32(trailer + 12, false);
+    const rootIndex = this.view.getUint32(trailer + 20, false);
+    let tableOffset = this.view.getUint32(trailer + 28, false);
+
+    // Parse all offsets before starting to parse objects
+    for (let i = 0; i < objectCount; i++) {
+      this.offsetTable.push(this.parseInteger(tableOffset, offsetSize));
+      tableOffset += offsetSize;
+    }
+
+    return this.parseObject(this.offsetTable[rootIndex]);
+  }
+
+  parseLengthAndOffset(offset: number, extra: number) {
+    if (extra !== 0x0f) return { length: extra, offset: 0 };
```

Comment:
please use brackets

Reply:
also please explain this condition, it's super unclear to me ! Maybe with a comment for this method too.